src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -571,26 +587,29 @@ private void handlePotentialStuckRepairs(LazyInitializer<Set<String>> busyHosts,
           LOG.debug("failed to query metrics for host {}, trying to get metrics from storage...", node, e);
         }
       }
-      return nodeDc.equals(localDc)
+
+      return !context.config.getDatacenterAvailability().isInCollocatedMode()
           ? Optional.empty()
           : getRemoteNodeMetrics(node, nodeDc);
     });
   }
 
   private Optional<NodeMetrics> getRemoteNodeMetrics(String node, String nodeDc) {
-    Preconditions.checkState(DatacenterAvailability.ALL != context.config.getDatacenterAvailability());
+    Preconditions.checkState(context.config.getDatacenterAvailability().isInCollocatedMode());
```

Comment:
shouldn't this be `Preconditions.checkState(!context.config.getDatacenterAvailability().isInCollocatedMode());` ?

Reply:
No, we want to make sure that we're only trying to grab metrics through the backend when we're in either `EACH` or `SIDECAR`.
The precondition looks good to me.

Reply:
Why can't `LOCAL` grab metrics?

Reply:
`LOCAL` will grab metrics directly from the nodes in the local DC. It won't go through the backend to get metrics for remote nodes that aren't directly reachable through JMX.

Reply:
The docs do state for `LOCAL` that `In addition, metrics can be collected asynchronously through the Apache Cassandra storage.`

see https://github.com/thelastpickle/cassandra-reaper/pull/622#discussion_r287893882

Reply:
Same as above, the code matches `LOCAL`'s current behavior in 1.4

Reply:
But it's not the behaviour in `master`.  The code was fixed intentionally in master, as 1.4 is identified as incorrect.

Similar to the decision in https://github.com/thelastpickle/cassandra-reaper/pull/622#discussion_r289596995
 this code should remain as it was in master beforehand, and we revisit the `LOCAL` vs `EACH` mode advantages outside of this PR.

Reply:
Thanks.